lib/matplotlib/axes/_axes.py

```diff
@@ -838,6 +838,75 @@ def axvline(self, x=0, ymin=0, ymax=1, **kwargs):
         self.autoscale_view(scalex=scalex, scaley=False)
         return l
 
+    @docstring.dedent_interpd
+    def axline(self, slope, intercept, **kwargs):
```

Comment:
Do we want to default the intercept to zero?

Reply:
Two additional thoughts:

- Is slope/intercept the common way, one wanted to use this, or would it be better to define the line by giving two points? The latter is a bit more general as you can also draw vertical lines that way (not that it's really needed because one can use `axvline`). I've no clear opinion on this myself.

- Maybe make the arguments keyword-only. `axline(0.5, 1)` is not very clear to read.

Reply:
Actually I think I like the "two points" parametrization.  Perhaps something like `axline(xy1, xy2=(0, 0), **kwargs)`?  so one can have lines passing by 0 with one argument, and other lines with two.

Reply:
If someone else wants to take this PR and change it to xy1 and xy2 feel free; I don't think I have the time/inclination to at the moment.

Reply:
I think most people using matplotlib can figure out the slope and intercept of a line, and since the point here is an infinite line, specifying two points would be a bit strange, and might lead folsk to think the line would be between the two points.  

OTOH, I like the idea of making kwarg only with intercept defaulting to zero...

Reply:
I'm coming around to xy1, xy2 now. I think it has the additional benifit that if you know `slope, intercept`, it's fairly easy to make up a `xy1, xy2`, but going the other way is a lot harder.

Reply:
The advantage of xy1, xy2 is that we could eventually refactor axvline to use this (to refactor those two, we'll have to support the two arguments that are not supported in axline right now). axvline would be possible with axline only if infinity slope is suported which is kind of weird.

Reply:
Agree with @NelleV about why (slope, intercept) is problematic (and it is the same reasons that the Hough transform uses (r, theta) (https://en.wikipedia.org/wiki/Hough_transform) ).